### PR TITLE
Function to strip solutions and interpolation of functions

### PR DESCRIPTION
**Pull request by jClugstor**
## Checklist

- [ ] Appropriate tests were added
- [ x] Any code changes were done in a way that does not break public API
- [x ] All documentation related to code changes were updated
- [x ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API
  
## Additional context

Adds a function to strip `InterpolationData` objects of their functions, and strip `ODEsolution` objects of their functions, in order to accommodate serialization better.

### CODE REVIEW

File: src/misc_utils.jl [hunk 1]
```diff
@@ -130,3 +130,16 @@ function get_differential_vars(f, u)
 end
 
 isnewton(::Any) = false
+
+function strip_solution(sol::ODESolution)
```
Comment (ChrisRackauckas) [inline src/misc_utils.jl:134]: This goes in SciMLBase, everything used here is defined there.

File: src/interp_func.jl [hunk 2]
```diff
@@ -59,3 +59,15 @@ function InterpolationData(id::InterpolationData, f)
         id.differential_vars,
         id.sensitivitymode)
 end
+
+# strip interpolation of function information
+function strip_interpolation(id::InterpolationData)
```
Comment (ChrisRackauckas) [inline src/interp_func.jl:64]: This needs to extend not shadow from SciMLBase, so make sure to import.

**Comment by ChrisRackauckas** [comment]
Rebase this PR onto https://github.com/SciML/OrdinaryDiffEq.jl/pull/2313, i.e. just add that one function

File: src/interp_func.jl (hunk 2 above)
Comment (jClugstor) [reply src/interp_func.jl:64]: is `SciMLBase.strip_interpolation ... ` fine, or should I do an explicit `import SciMLBase: strip_interpolation

Comment (ChrisRackauckas) [reply src/interp_func.jl:64]: namespacing is fine.

**Pull request closed without merging**